As expected, this test successfully fails in CI:

```
	/usr/local/go/src/testing/testing.go:960 +0x350

goroutine 1 [chan receive]:
testing.(*T).Run(0xc00022ce00, 0x119e92f, 0x1b, 0x121ff08, 0x484c01)
	/usr/local/go/src/testing/testing.go:961 +0x377
testing.runTests.func1(0xc00022c500)
	/usr/local/go/src/testing/testing.go:1202 +0x78
testing.tRunner(0xc00022c500, 0xc000305d88)
	/usr/local/go/src/testing/testing.go:909 +0xc9
testing.runTests(0xc0002a4660, 0x1b2bac0, 0x23, 0x23, 0x0)
	/usr/local/go/src/testing/testing.go:1200 +0x2a7
testing.(*M).Run(0xc00026f400, 0x0)
	/usr/local/go/src/testing/testing.go:1117 +0x176
main.main()
	_testmain.go:184 +0x1c1

goroutine 19 [chan receive]:
k8s.io/klog.(*loggingT).flushDaemon(0x1b3a5e0)
	/go/pkg/mod/k8s.io/klog@v0.4.0/klog.go:1018 +0x8b
created by k8s.io/klog.init.0
	/go/pkg/mod/k8s.io/klog@v0.4.0/klog.go:404 +0x6c
FAIL	helm.sh/helm/v3/pkg/chartutil	3.157s
FAIL
make: *** [Makefile:81: test-coverage] Error 1
```

--- a/pkg/chartutil/errors.go
+++ b/pkg/chartutil/errors.go
@@ -21,11 +21,15 @@ import (
 )
 
 // ErrNoTable indicates that a chart does not have a matching table.
-type ErrNoTable string
+type ErrNoTable struct {
+	Key string
+}
 
-func (e ErrNoTable) Error() string { return fmt.Sprintf("%q is not a table", e) }
+func (e ErrNoTable) Error() string { return fmt.Sprintf("%q is not a table", e.Key) }
 
 // ErrNoValue indicates that Values does not contain a key with a value
-type ErrNoValue string
+type ErrNoValue struct {
+	Key string
+}
 
-func (e ErrNoValue) Error() string { return fmt.Sprintf("%q is not a value", e) }
+func (e ErrNoValue) Error() string { return fmt.Sprintf("%q is not a value", e.Key) }

--- a/pkg/chartutil/errors_test.go
+++ b/pkg/chartutil/errors_test.go
@@ -1,5 +1,3 @@
-package chartutil
-
 /*
 Copyright The Helm Authors.
 
@@ -16,28 +14,24 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+package chartutil
+
 import (
 	"testing"
 )
 
 func TestErrorNoTableDoesntPanic(t *testing.T) {
-
-	defer recover()
-
 	x := "empty"
 
-	y := ErrNoTable(x)
+	y := ErrNoTable{x}
 
 	t.Logf("error is: %s", y)
 }
 
 func TestErrorNoValueDoesntPanic(t *testing.T) {
-
-	defer recover()
-
 	x := "empty"
 
-	y := ErrNoValue(x)
+	y := ErrNoValue{x}
 
 	t.Logf("error is: %s", y)
 }

--- a/pkg/chartutil/values.go
+++ b/pkg/chartutil/values.go
@@ -88,7 +88,7 @@ func (v Values) Encode(w io.Writer) error {
 func tableLookup(v Values, simple string) (Values, error) {
 	v2, ok := v[simple]
 	if !ok {
-		return v, ErrNoTable(simple)
+		return v, ErrNoTable{simple}
 	}
 	if vv, ok := v2.(map[string]interface{}); ok {
 		return vv, nil
@@ -101,7 +101,7 @@ func tableLookup(v Values, simple string) (Values, error) {
 		return vv, nil
 	}
 
-	return Values{}, ErrNoTable(simple)
+	return Values{}, ErrNoTable{simple}
 }
 
 // ReadValues will parse YAML byte data into a Values.
@@ -193,20 +193,20 @@ func (v Values) pathValue(path []string) (interface{}, error) {
 		if _, ok := v[path[0]]; ok && !istable(v[path[0]]) {
 			return v[path[0]], nil
 		}
-		return nil, ErrNoValue(path[0])
+		return nil, ErrNoValue{path[0]}
 	}
 
 	key, path := path[len(path)-1], path[:len(path)-1]
 	// get our table for table path
 	t, err := v.Table(joinPath(path...))
 	if err != nil {
-		return nil, ErrNoValue(key)
+		return nil, ErrNoValue{key}
 	}
 	// check table for key and ensure value is not a table
 	if k, ok := t[key]; ok && !istable(k) {
 		return k, nil
 	}
-	return nil, ErrNoValue(key)
+	return nil, ErrNoValue{key}
 }
 
 func parsePath(key string) []string { return strings.Split(key, ".") }

